tests/test_factory.py

```diff
@@ -288,3 +288,57 @@ def test_topology_count_collapse():
         )
     )
     assert dag.count_topologies(collapse_leaves=True) == 2
+
+
+# this tests is each of the trees indexed are valide subtrees
+# they should have exactly one edge descending from each node clade pair
+def test_valid_subtrees():
+    for history_dag in dags + cdags:
+        # get the set of all dags that were indexed
+        # all_dags_indexed = {None}  # set of all the indexed dags
+        # curr_dag_index = 0
+        # while not history_dag[curr_dag_index] is None:
+        for curr_dag_index in range(0, len(history_dag)):
+            next_tree = history_dag[curr_dag_index]
+            # print("in the while loop")
+            # print(next_tree)
+            # print(next_tree.to_newick())
+            assert next_tree.to_newick() in history_dag.to_newicks()
+            assert next_tree.is_clade_tree()
+            # curr_dag_index = curr_dag_index + 1
+
+
+# this should check if the indexing algorithm accurately
+# captures all possible subtrees of the dag
+def test_indexing_comprehensive():
+    for history_dag in dags + cdags:
+        # get the set of all dags that were indexed
+        all_dags_indexed = {None}  # set of all the indexed dags
+
+        # get all the possible dags using indexingd
+        for curr_dag_index in range(0, len(history_dag)):
+            next_tree = history_dag[curr_dag_index]
+            all_dags_indexed.add(next_tree.to_newick())
+        print("len: " + str(len(history_dag)))
+        print("number of indexed trees: " + str(len(all_dags_indexed)))
+
+        # get the set of all actual dags from the get_trees
+        all_dags_true_generator = history_dag.to_newicks()
+        all_dags_true = {None}
```

Comment:
I think part of this test could be simplified with something like:
`assert set(dag.to_newicks()) == {tree.to_newick() for tree in dag}`